Media self sending broken

Occurs with latest HEAD.
I suspect #2237 causes this, as it worked before. Could be coincidence.

Self sending media message leads to
- receiving message bubble misses image 
- the avater is default mr. open mouth (though contact has an own avatar)
- receive bubble only appears on reentering conversation or sending another message
- sometimes the bubble is green instead of blue after sending (blue after reentering)

Text only works though - avatar is correct. Also receiving non self messages seems to work.

```
12-23 10:21:26.172  13028-13054/org.thoughtcrime.securesms W/MmsSmsDatabase﹕ Executing query: SELECT _id, body, type, thread_id, address, address_device_id, subject, date_sent, date_received, m_type, msg_box, status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (thread_id = 11) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (thread_id = 11) ORDER BY date_received ASC)
12-23 10:21:26.182  13028-13846/org.thoughtcrime.securesms W/MmsSmsDatabase﹕ Executing query: SELECT _id, body, type, thread_id, address, address_device_id, subject, date_sent, date_received, m_type, msg_box, status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, transport_type FROM (SELECT DISTINCT date_sent * 1 AS date_sent, date * 1 AS date_received, _id, body, read, thread_id, type, address, address_device_id, subject, NULL AS m_type, NULL AS msg_box, status, NULL AS part_count, NULL AS ct_l, NULL AS tr_id, NULL AS m_size, NULL AS exp, NULL AS st, delivery_receipt_count, 'sms' AS transport_type FROM sms WHERE (thread_id = 11) UNION ALL SELECT DISTINCT date * 1000 AS date_sent, date_received * 1000 AS date_received, _id, body, read, thread_id, NULL AS type, address, address_device_id, NULL AS subject, m_type, msg_box, NULL AS status, part_count, ct_l, tr_id, m_size, exp, st, delivery_receipt_count, 'mms' AS transport_type FROM mms WHERE (thread_id = 11) ORDER BY date_received ASC)
12-23 10:21:26.182  13028-13063/org.thoughtcrime.securesms W/PartDatabase﹕ Getting part at: /data/data/org.thoughtcrime.securesms/app_parts/part266234640.mms
12-23 10:21:26.222  13028-13063/org.thoughtcrime.securesms W/PartDatabase﹕ Getting part at: /data/data/org.thoughtcrime.securesms/app_parts/part266234640.mms
12-23 10:21:26.262  13028-13028/org.thoughtcrime.securesms W/ImageSlide﹕ Got soft reference: null
12-23 10:21:26.282  13028-13028/org.thoughtcrime.securesms W/ImageSlide﹕ Got soft reference: java.lang.ref.SoftReference@434d85b8
12-23 10:21:26.282  13028-13028/org.thoughtcrime.securesms W/ImageSlide﹕ Got cached bitmap: android.graphics.drawable.BitmapDrawable@42f6e0b8
12-23 10:21:26.282  13028-13028/org.thoughtcrime.securesms W/ImageSlide﹕ Setting cached thumbnail...
12-23 10:21:26.402  13028-13060/org.thoughtcrime.securesms W/PartDatabase﹕ inserting part to mms 93
12-23 10:21:26.402  13028-13060/org.thoughtcrime.securesms W/PartDatabase﹕ Writing part data from URI
12-23 10:21:26.402  13028-13060/org.thoughtcrime.securesms E/ActivityThread﹕ Failed to find provider info for org.thoughtcrime.securesms
12-23 10:21:26.432  13028-13060/org.thoughtcrime.securesms W/MessageSender﹕ ws.com.google.android.mms.MmsException: java.io.FileNotFoundException: No content provider: content://org.thoughtcrime.securesms/part/114
            at org.thoughtcrime.securesms.database.PartDatabase.writePartData(PartDatabase.java:354)
            at org.thoughtcrime.securesms.database.PartDatabase.insertPart(PartDatabase.java:376)
            at org.thoughtcrime.securesms.database.PartDatabase.insertParts(PartDatabase.java:184)
            at org.thoughtcrime.securesms.database.MmsDatabase.insertMediaMessage(MmsDatabase.java:728)
            at org.thoughtcrime.securesms.database.MmsDatabase.copyMessageInbox(MmsDatabase.java:529)
            at org.thoughtcrime.securesms.sms.MessageSender.sendMediaSelf(MessageSender.java:175)
            at org.thoughtcrime.securesms.sms.MessageSender.sendMediaMessage(MessageSender.java:136)
            at org.thoughtcrime.securesms.sms.MessageSender.send(MessageSender.java:105)
            at org.thoughtcrime.securesms.ConversationActivity$15.doInBackground(ConversationActivity.java:1070)
            at org.thoughtcrime.securesms.ConversationActivity$15.doInBackground(ConversationActivity.java:1067)
            at android.os.AsyncTask$2.call(AsyncTask.java:288)
            at java.util.concurrent.FutureTask.run(FutureTask.java:237)
            at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
     Caused by: java.io.FileNotFoundException: No content provider: content://org.thoughtcrime.securesms/part/114
            at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1049)
            at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method)
            at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:631)
            at android.content.ContentResolver.openTypedAssetFileDescriptor(Native Method)
            at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:904)
            at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method)
            at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:631)
            at android.content.ContentResolver.openAssetFileDescriptor(Native Method)
            at android.content.ContentResolver.openInputStream(ContentResolver.java:629)
            at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method)
            at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:631)
            at android.content.ContentResolver.openInputStream(Native Method)
            at org.thoughtcrime.securesms.database.PartDatabase.writePartData(PartDatabase.java:348)
            at org.thoughtcrime.securesms.database.PartDatabase.insertPart(PartDatabase.java:376)
            at org.thoughtcrime.securesms.database.PartDatabase.insertParts(PartDatabase.java:184)
            at org.thoughtcrime.securesms.database.MmsDatabase.insertMediaMessage(MmsDatabase.java:728)
            at org.thoughtcrime.securesms.database.MmsDatabase.copyMessageInbox(MmsDatabase.java:529)
            at org.thoughtcrime.securesms.sms.MessageSender.sendMediaSelf(MessageSender.java:175)
            at org.thoughtcrime.securesms.sms.MessageSender.sendMediaMessage(MessageSender.java:136)
            at org.thoughtcrime.securesms.sms.MessageSender.send(MessageSender.java:105)
            at org.thoughtcrime.securesms.ConversationActivity$15.doInBackground(ConversationActivity.java:1070)
            at org.thoughtcrime.securesms.ConversationActivity$15.doInBackground(ConversationActivity.java:1067)
            at android.os.AsyncTask$2.call(AsyncTask.java:288)
            at java.util.concurrent.FutureTask.run(FutureTask.java:237)
            at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
12-23 10:21:26.442  13028-13063/org.thoughtcrime.securesms W/BitmapUtil﹕ rough scale 720x1280 => 360x640
12-23 10:21:26.442  13028-13063/org.thoughtcrime.securesms W/BitmapUtil﹕ fine scale  360x640 => 146.8125x261.0
12-23 10:21:26.452  13028-13063/org.thoughtcrime.securesms I/CameraExif﹕ Orientation not found
12-23 10:21:26.452  13028-13063/org.thoughtcrime.securesms W/ImageSlide﹕ Got soft reference: java.lang.ref.SoftReference@430073c0
12-23 10:21:26.452  13028-13063/org.thoughtcrime.securesms W/ImageSlide﹕ Got cached bitmap: android.graphics.drawable.BitmapDrawable@427b4148
```

--- a/src/org/thoughtcrime/securesms/database/PartDatabase.java
+++ b/src/org/thoughtcrime/securesms/database/PartDatabase.java
@@ -345,7 +345,7 @@ public class PartDatabase extends Database {
         return writePartData(masterSecret, part, new ByteArrayInputStream(part.getData()));
       } else if (part.getDataUri() != null) {
         Log.w(TAG, "Writing part data from URI");
-        InputStream in = context.getContentResolver().openInputStream(part.getDataUri());
+        InputStream in = PartAuthority.getPartStream(context, masterSecret, part.getDataUri());
         return writePartData(masterSecret, part, in);
       } else {
         throw new MmsException("Part is empty!");

